src/core/project/qgsproject.h

```diff
@@ -862,20 +863,35 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
     QgsProjectElevationProperties *elevationProperties();
 
     /**
-     * Returns the project's display settings, which settings and properties relating
+     * Returns the project's display settings, which contains settings and properties relating
      * to how a QgsProject should display values such as map coordinates and bearings.
      * \note not available in Python bindings
      * \since QGIS 3.12
      */
     const QgsProjectDisplaySettings *displaySettings() const SIP_SKIP;
 
     /**
-     * Returns the project's display settings, which settings and properties relating
+     * Returns the project's display settings, which contains settings and properties relating
      * to how a QgsProject should display values such as map coordinates and bearings.
      * \since QGIS 3.12
      */
     QgsProjectDisplaySettings *displaySettings();
 
+    /**
+     * Returns the project's GPS settings, which contains settings and properties relating
+     * to how a QgsProject should interact with a GPS device.
+     * \note not available in Python bindings
+     * \since QGIS 3.30
+     */
+    const QgsProjectGpsSettings *gpsSettings() const SIP_SKIP;
```

Comment:
I'm wondering here if we shouldn't just go ahead wiht gnssSettings() instead since it's a completely new concept being introduced. Or we just leave it as gps knowing we'll break that soon enough in 4.0. Not sure what's best between consistency vs. introducing an API that won't break in 1 year time. Truly 50/50 on this so go with your gut :)

Reply:
I think we stick with GPS for 3.x... it's just too unpredictable otherwise